src/Symfony/Component/Lock/Store/ZookeeperStore.php

```diff
@@ -0,0 +1,239 @@
+        // This makes sure the node dies at the end of a session, thus releasing the lock to avoid starvation
+        $flags = Zookeeper::EPHEMERAL;
+
+        try {
+            $token = $this->getToken($key);
+            $this->createNewLock($lockNodePath, $token, array(), $flags);
+        } catch (LockConflictedException $ex) {
+            throw $ex;
+        } catch (Throwable $throwable) {
+            throw new LockAcquiringException($throwable);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws LockReleasingException
+     */
+    public function delete(Key $key)
+    {
+        $lockNodePath = self::LOCK_NODE.'/'.$this->getKeyResource($key);
+        try {
+            if ($this->exists($key)) {
+                $this->zookeeper->delete($lockNodePath);
+            }
+        } catch (Throwable $throwable) {
+            // For Zookeeper Ephemeral Nodes, the node will be deleted upon session death. But, if we want to unlock
+            // the lock before proceeding further in the session, the client should be aware of this
+            throw new LockReleasingException($throwable);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function exists(Key $key): bool
+    {
+        $lockNodePath = self::LOCK_NODE.'/'.$this->getKeyResource($key);
+        if (!$this->zookeeper->exists($lockNodePath)) {
+            return false;
+        }
+
+        return $this->zookeeper->get($lockNodePath) === $this->getToken($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws NotSupportedException
+     */
+    public function waitAndSave(Key $key)
+    {
+        throw new NotSupportedException();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function putOffExpiration(Key $key, $ttl)
+    {
+    }
+
+    /**
+     * @param int    $errNo  Error number
+     * @param string $errStr Error String
+     *
+     * @throws LockConflictedException
+     * @throws LockAcquiringException
+     */
+    public static function errorHandlerForCreateNode($errNo, $errStr)
+    {
+        if (E_WARNING === $errNo && 'Zookeeper::create(): error: node exists' === $errStr) {
+            throw new LockConflictedException($errStr);
+        }
+
+        throw new LockAcquiringException($errStr);
+    }
+
+    /**
+     * Create a zookeeper node.
+     * If it has a path that doesn't exists, then this method creates the path with the same permissions before
+     * creating a node.
+     *
+     * @param string $node  The node which needs to be created
+     * @param string $value The value to be assigned to a zookeeper node
+     * @param array  $acl   Permissions settings on a zookeeper node
+     * @param int    $type  Type of zookeeper node OPTIONAL
+     *
+     * @return string Path of the node created
+     *
+     * @throws LockConflictedException
+     * @throws LockAcquiringException
+     */
+    private function createNewLock(
```

Comment:
this private method is used once, IMO you can remove it

Reply:
I understand it's not being used more than once, but I feel it improves readability of the code and abstracts out the functionality well. Unless you have strong objection I am going to leave it as is.